File: CRUD-Operations-Angular/ClientApp/app/components/models/projectViewModel.ts
import { UsersProjectsViewModel } from "./usersProjectsViewModel";
export class ProjectViewModel
{
    public id: number;
    public name: string;
    public description: string;

    public startDay: number;
    public startMonth: number;
    public startYear: number;

    public endDay: number;
    public endMonth: number;
    public endYear: number;
    public users: UsersProjectsViewModel[];

    public static GetProject(name: string, description: string, startDay: number, startMonth: number,
        startYear: number, endDay: number, endMonth: number, endYear: number): ProjectViewModel
        {
            let result: ProjectViewModel = new ProjectViewModel();
            result.name = name;
            result.description = description;
            result.startDay = startDay;
            result.startMonth = startMonth;
            result.startYear = startYear;
            result.endDay = endDay;
            result.endMonth = endMonth;
            result.endYear = endYear;
            return result;
        }

    public constructor(){}
    // public constructor(name: string, description: string,
    //     startDay: number, startMonth: number, startYear: number,
    //     endDay: number, endMonth: number, endYear: number, users: UsersProjectsViewModel[])
    // {
    //     this.Name = name;
    //     this.Description = description;
    //     this.StartDay = startDay;
    //     this.StartMonth = startMonth;
    //     this.StartYear = startYear;
    //     this.EndDay = endDay;
    //     this.EndMonth = endMonth;
    //     this.EndYear = endYear;
    //     this.Users = users;
    // }

    
}

File: CRUD-Operations-Angular.DataAccess/ViewModels/ProjectViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CRUD_Operations_Angular.DataAccess.ViewModels
{
    public class ProjectViewModel: IViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public int StartDay { get; set; }
        public int StartMonth { get; set; }
        public int StartYear { get; set; }

        public int EndDay { get; set; }
        public int EndMonth { get; set; }
        public int EndYear { get; set; }
        public ICollection<UsersProjectsViewModel> Users { get; set; }
    }
}
